contracts/Exclusivity.vy

```diff
@@ -1,90 +1,141 @@
 # @version ^0.3.7
-# this relies on functional ActiveUser contract/class
-# no other contract is interfacing with this code as of 12/12 at 9:00AM
+# relies on only Lottery contract as of June 5 2023
+
 interface ActiveUser:
     def getIsActiveUser(potentialUser: address) -> bool: view
     def getIsAdmin(potentialAdmin: address) -> bool: view
-    
+
+interface Lottery:
+    def getPot() -> uint256: view
+    def setStartingPot(amount: uint256) : nonpayable
+
 activeUserContract: public(ActiveUser)
 
+lotteryContract: public(Lottery)
+
 admin: HashMap[address, bool]
+votingAddress: address
+rickyCWallet: address
 topicsAddress: public(DynArray[address, 1000])
-percentage: uint256
+percentage: uint256 
 classSize: uint256
+activeStudents: HashMap[address, uint256]
+
 @external
-def vote(voter: address):
-    isIn: bool = False
-    for studentAddress in self.topicsAddress: #find index of address of candidate in topics addresses
-            if studentAddress==voter:    
-                isIn = True
-                break
-    if isIn == True:
-        self._removeNonTopics(voter)
-        send(voter,1)
-    if self.admin[voter]:
-     
-        send(voter,(15/100)*self.classSize)
+def __init__(erc20address: address):
+    self.lotteryContract = Lottery(erc20address)
+    self._setRickyC()
+
+# Function 1: students can use Wolvercoin to vote for initiatives; Mr. Theiss's vote can be weighted up to 15% of the total vote:
```

Comment:
Instead of MrTheiss's vote, it should check for an 'admin' using the Active User contract (instead of self._checkAdmin)